### PR TITLE
Use the C# identifier from the calling code in the failure message instead of a generic name

### PR DESCRIPTION
**Pull request by dennisdoomen**
See #588 

*Order of precedence*
1. Context (for nested object graph assertion)
2. Identifier extracted from the calling source code
3. Fallback identifier specified in the `{identifier:fallback}` placeholder
4. Generic fallback provided through `WithDefaultIdentifier`

### CODE REVIEW

File: Src/FluentAssertions/CallerIdentifier.cs [hunk 1]
```diff
@@ -69,6 +70,16 @@ private static string ExtractVariableNameFrom(StackFrame frame)
             return caller;
         }
 
+        private static bool UsesNewKeyword(string candidate)
+        {
+            return new Regex(@"new(\s?\[|\s?\{|\s\w+)").IsMatch(candidate);
```
Comment (jnyrup) [inline Src/FluentAssertions/CallerIdentifier.cs]: Just nitpicking, but the static [`Regex.IsMatch`](https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regex.ismatch?view=netframework-4.7.1#System_Text_RegularExpressions_Regex_IsMatch_System_String_System_String_) caches the regex.

Comment (dennisdoomen) [reply Src/FluentAssertions/CallerIdentifier.cs]: Good point. I still have to clean up the code. Currently focusing on getting all tests green.

**Comment by jnyrup** [comment]
Is `CallerIdentifier.logger` intentional or an just oversight before committing?

**Comment by dennisdoomen** [comment]
I used it for diagnostic purposes and plan to keep it in in case it misbehaves. Have to think of a way to make it available as part of the tracing functionality of the `AssertionScope`.

**Comment by jnyrup** [comment]
Ok. 
I mostly just noticed that there was a public static lowercase field in a public class, which was never reassigned.

**Comment by dennisdoomen** [comment]
Yeah, thanks for reminding me about that one.

**Pull request merged**